File: lab2-functionalprimer/python3/func_as_arg.py
# Function to generate a list of integers from a to b
def listFunc(a, b):
    return [i for i in range(a, b + 1)]  # Equivalent to Haskell's [a..b]

# Function to compute sum or average
def applicatorFunc(inpFunc, a, b, s):
    numbers = inpFunc(a, b)  # Generate list using inpFunc
    return sum(numbers) if s == 's' else sum(numbers) / len(numbers)  # Sum or Average

# Take user input
a = int(input("Enter the starting number (a): "))
b = int(input("Enter the ending number (b): "))
s = input("Enter 's' for sum or any other character for average: ")

# Call the function and print the result
result = applicatorFunc(listFunc, a, b, s)
print("Result =", result)

File: lab2-functionalprimer/javascript/func_as_arg.js
// Function to generate an array of integers from a to b
function arrFunc(a, b) {
    let arr = [];
    for (let i = a; i <= b; i++) {
        arr.push(i);
    }
    return arr;
}

// Function to compute sum or average
function applicatorFunc(inpFunc, a, b, s) {
    const arr = inpFunc(a, b); // Generate array using inpFunc
    let sum = arr.reduce((acc, val) => acc + val, 0);
    return s === 's' ? sum : sum / arr.length; // Compute sum or average
}

// Take user input
const a = parseInt(prompt("Enter the starting number (a):"));
const b = parseInt(prompt("Enter the ending number (b):"));
const s = prompt("Enter 's' for sum or any other character for average:");

// Call the function and display the result
const result = applicatorFunc(arrFunc, a, b, s);
console.log("Result =", result);
